### PR TITLE
Lots of different changes!

### PR DESCRIPTION
**Pull request by sergeyf**
Fixes:

(1) My previous change for fixing author_info_middle bug had a little bug in it. Fixed.
(2) New logic for names to take care of the issue where Chris/Le/Murdy was looking different than Chris//Le Murdy. This requires a change in how one of the features is computed to make it looser.
(3) Before the feature for "same_publisher_sources" was looking at basically ALL publishers. That's changed to a subset where the publishers are known to be high quality.
(4) The word "signatures" is left-over S2AND. I renamed that in model.py
(5) Fix for null titles: they are now removed before clustering, then added back in based on IDs.
(6) Post-hoc joining of clusters wherever there are overlaps in DOI/pdf_hash/PMID.
(7) Text normalization has a new exception for '-'. This is to turn names like Jo-Ann into JoAnn instead of Jo Ann.

### CODE REVIEW

**Review by stefanc-ai2 — APPROVED** [review]

File: scripts/dump_pairwise_model.py [hunk 1]
```diff
@@ -148,5 +161,5 @@
 models = {}
 models["clusterer"] = cluster
 
-with open(join(CONFIG["main_data_dir"], "prod_model.pickle"), "wb") as f:
+with open(join(CONFIG["main_data_dir"], "prod_model_0.1.4.pickle"), "wb") as f:
```
Comment (stefanc-ai2) [inline scripts/dump_pairwise_model.py:164]: Need to include this new pickle file as part of the pull request?

**Review by stefanc-ai2 — APPROVED** [review]
Just the thing about including the new pickle file. Otherwise, LGTM!

File: scripts/dump_pairwise_model.py (hunk 1 above)
Comment (sergeyf) [reply scripts/dump_pairwise_model.py:164]: I renamed it to `prod_model.pickle`, put it in the `artifacts` folder (not part of github repo) and ran `tt publish-artifacts --artifacts-dir artifacts --s3-path s3://ai2-s2-research-public/paper_clustering_artifacts_v0_2.tar.gz`

Comment (sergeyf) [reply scripts/dump_pairwise_model.py:164]: Is that sufficient? That's the pickle specified in the new variant in the config.

**Pull request merged**